posthog/hogql/functions/action.py

```diff
@@ -0,0 +1,45 @@
+from typing import List
+
+from posthog.hogql import ast
+from posthog.hogql.context import HogQLContext
+from posthog.hogql.errors import QueryError
+from posthog.hogql.escape_sql import escape_clickhouse_string
+
+
+def matches_action(node: ast.Expr, args: List[ast.Expr], context: HogQLContext) -> ast.Expr:
+    arg = args[0]
+    if not isinstance(arg, ast.Constant):
+        raise QueryError("action() takes only constant arguments", node=arg)
+    if context.team_id is None:
+        raise QueryError("action() can only be used in a query with a team_id", node=arg)
+
+    from posthog.models import Action
+    from posthog.hogql.property import action_to_expr
+
+    if (isinstance(arg.value, int) or isinstance(arg.value, float)) and not isinstance(arg.value, bool):
+        actions = Action.objects.filter(id=int(arg.value), team_id=context.team_id).all()
+        if len(actions) == 1:
+            context.add_notice(
+                start=arg.start,
+                end=arg.end,
+                message=f"Action #{actions[0].pk} can also be specified as {escape_clickhouse_string(actions[0].name)}",
```

Comment:
What's the purpose of this notice? The other notice makes sense as it acts like a warning, but if people followed this notice they would run into the other one. Seems like the notices should guide people into doing things the "best" way.

I can see we have this pattern in posthog/hogql/functions/cohort.py so maybe I am just missing the point of these notices 🙃

Reply:
With cohorts we give you the option of referring to them by ID (stability) or searching by name (flexibility). I just took the same approach. I assume most users will search via name, but some might not. Since I don't know what anyone will prefer, I didn't want to force users down one path... 🤷